jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererRecommendationManager.kt

```diff
@@ -135,13 +135,13 @@ class CodeWhispererRecommendationManager {
             val truncated = it.toBuilder()
                 .content(truncatedContent)
                 .build()
-            val isDiscardedByUserInputForTruncated = !truncated.content().startsWith(userInput)
+            val isDiscardedByUserInputForTruncated = !truncated.content().startsWith(userInput) || truncated.content() == userInput
```

Comment:
so right now, the bug case you encountered will be captured by this statement

Reply:
should I remove `tuncatedContent == userInput` at  L144 ?

Reply:
The condition `truncated.content().isEmpty()` is safe to remvoe right? So if I understand it correctly, this one is supposed to be the one I just added (`truncatedContent == userInput`)?

Reply:
if truncated is empty, `!truncated.content().startsWith(userInput) || truncated.content() == userInput` this must be true

Reply:
I see